### ISSUE TITLE
ip6tnl0 interface not up when systemd-networkd is restarted

### ISSUE THREAD

**Issue by anitazha** [systemd/systemd#25927]
### systemd version the issue has been seen with

251

### Used distribution

CentOS Stream 9

### Linux kernel version used

5.12.0-0

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

I have this network config:
```
# cat /etc/systemd/network/50-fb_networkd-ip6tnl0.network 
[Network]
Address = <IP address omitted>

[Match]
Name = ip6tnl0
```
If I reboot the host, I expect systemd-networkd to bring up the interface.

### Unexpected behaviour you saw

Given the network config above, I expect systemd-networkd to bring up ip6tnl0 interface when it starts. However, systemd-networkd does not do so until I explicitly run `networkctl reconfigure ip6tnl0`.

I realize I am relying on some implicit behavior here, such as the kernel automagically loading the ip6_tunnel module when the `ip6tnl0` interface is added. However it seems odd to me that just starting systemd-networkd on a fresh reboot and explicitly running `networkctl reconfigure ip6tnl0` do not produce the same results.

### Steps to reproduce the problem

- Create the config as above with an IP address
- Run `systemctl restart systemd-networkd` and notice that `ip6tnl0` is not up after checking `ip a`
- Run `networkctl reconfigure ip6tnl0` and notice that `ip6tnl0` is up after checking `ip a`

**Comment by yuwata** [comment]
Sorry for late response.

Let me confirm, that the tunnel is not create by a .netdev file, but automatically created by kernel when the corresponding module is loaded. Right?

**Comment by anitazha** [comment]
> Let me confirm, that the tunnel is not create by a .netdev file, but automatically created by kernel when the corresponding module is loaded. Right?

That's right, we do not have a .netdev file and we expect the kernel to create the the tunnel and interface when the module is loaded. Manually loading the module creates `ip6tnl0` so I suspect there is something different in the `networkctl reconfigure` vs the networkd restart flow.
